### PR TITLE
[enh] Add Problem Matcher

### PR DESCRIPTION
**Pull request by dgautherot**
Hello,

I was about to create an Ada language support extension for Visual Studio Code when I discovered your repo and I thing you are on the right way by using the Language Server Protocol.

I would love to contribute to your work with this first pull request, adding a problem matcher for this extension.

Would it be possible to integrate it an push a new version of the Visual Studio Extension onto its marketplace ?

Thanks a lot.
Kind regard,

Denis Gautherot.

### CODE REVIEW

**Comment by setton** [comment]
Thank you @dgautherot! We'll take a look ASAP.

File: integration/vscode/ada/package.json [hunk 1]
```diff
@@ -99,6 +99,23 @@
...
             }
+        ],
+        "problemMatchers": [
+            {
+                "name": "ada",
+                "label": "Gnat Ada Problem Matcher",
+                "fileLocation": ["relative", "${workspaceRoot}"],
+                "pattern": [
+                    {
+                        "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|)(?:\\s+)?(.*)$",
```
Comment (reznikmm) [inline integration/vscode/ada/package.json]: I'm afraid first part of the regexp won't work on Windows. (Windows has drive letters, like `C:\`). I propose replace `^(.*)` with something like this `^([^:]:?[^:]*)`

What do you think?

Comment (dgautherot) [reply integration/vscode/ada/package.json]: Hello reznikmm!
I'm fine with your proposal!
So here would be the final regex (i've tested with specific cases and noticed a little typo with the final ':' that was included into the message part):

"^([^:]:?[^:]*):(\\d+):(\\d+):\\s+(warning|)(?:[:]\\s+)?(.*)$"

**Comment by reznikmm** [comment]
@dgautherot New version of the extension (v21.0.2) is on the marketplace! Thank you for contributing!

**Pull request merged**